src/jobs/config_sync_job.ts

```diff
@@ -377,6 +489,133 @@ export class ConfigSyncJob extends SyncJob {
+              new MappingsObject(
+                secondObject.id,
+                secondObject.name,
+                first ? this._connection.secondService.name : this._connection.firstService.name,
+                secondObject.type)
+            mapping.mappingsObjects.push(secondMappingsObject)
+            newMappings.push(mapping)
+          }
+        }
+      }
+    }
+    return newMappings
+  }
+
+  private async _deleteObsoleteMappingsAndTESOs(mappingsToDelete: Mapping[]): Promise<boolean> {
+    let success = true
+    const timeEntriesToArchive: TimeEntrySyncedObject[] = new Array()
+    //find TESOs to Archive
+    for (let mapping of mappingsToDelete) {
+      const firstService = SyncedServiceCreator.create(this._connection.firstService)
+      const secondService = SyncedServiceCreator.create(this._connection.secondService)
+      const TESOs: TimeEntry[] = new Array()
+      const TESOsFromFirst = await firstService.getTimeEntriesRelatedToMappingObjectForConnection(mapping, this._connection);
+      const TESOsFromSecond = await secondService.getTimeEntriesRelatedToMappingObjectForConnection(mapping, this._connection);
+      if (TESOsFromFirst)
+        TESOs.push(...TESOsFromFirst)
+      if (TESOsFromSecond)
+        TESOs.push(...TESOsFromSecond)
+      if (TESOs) {
+        for (let i = 0; i < TESOs.length; i++) {
+          const timeEntryFromApi = TESOs[i]
+          const firstOrSecond = TESOsFromFirst ? i < TESOsFromFirst.length : false
+          const foundTESO = await databaseService.getTimeEntrySyncedObjectForArchiving(
+            timeEntryFromApi.id,
+            firstOrSecond ? this._connection.firstService.name : this._connection.secondService.name,
+            this._user._id);
+          if (foundTESO) {
+            timeEntriesToArchive.push(foundTESO);
+          }
+        }
+      } else {
+        success = false
+      }
+    }
+    //deleteTESOs
+    for (const timeEntryToArchive of timeEntriesToArchive) {
+      const updateResponse = await databaseService.makeTimeEntrySyncedObjectArchived(timeEntryToArchive);
+      if (!updateResponse) {
+        success = false
+      }
+    }
+    //delete from service
+    const firstSyncedService = SyncedServiceCreator.create(this._connection.firstService)
+    const secondSyncedService = SyncedServiceCreator.create(this._connection.secondService)
+    mappingsToDelete.forEach(mapping => {
+      const secondaryMappingObject = mapping.mappingsObjects[0].id === mapping.primaryObjectId ? mapping.mappingsObjects[0] : mapping.mappingsObjects[1]
+      const syncedService = secondaryMappingObject.name === this._connection.firstService.name ? firstSyncedService : secondSyncedService
+      try {
+        syncedService.deleteServiceObject(secondaryMappingObject.id, secondaryMappingObject.type)
+      } catch (ex) {
+        //this can fail in case wrong service is called. (Jira and Redmine do not supprot deleting objects)
+      }
```

Comment:
Nešlo by tohle vyřešit nějak lépe? Třeba u mappingsObject mít informaci o service?

Reply:
Hledám druhý oběkt ze spojení. ten neukládám do mappings object.